### PR TITLE
Basic auth

### PR DESCRIPTION
**Pull request by KitaitiMakoto**
Hi,

Because `groonga-httpd` supports HTTP Basic Authentication, it is nice if groonga-client also do, isn't it?

### CODE REVIEW

**Comment by cosmo0920** [comment]
Looks good. But Travis CI fails except for Ruby 2.0.0 job....
Could you rebase at master after #5 is merged?

**Comment by KitaitiMakoto** [comment]
Okay.
Will do tonight or tomorrow.

File: lib/groonga/client.rb [hunk 1]
```diff
@@ -33,6 +33,10 @@ class << self
       #   @param [Hash] options The options.
       #   @option options [:gqtp or :http] :protocol The protocol that is
       #     used by the client.
+      #   @option options [String] :auth_user User ID for HTTP Basic
```
Comment (kou) [inline lib/groonga/client.rb]: How about removing `auth_` prefix? Because we don't have any other use cases for user and password in client.

**Comment by KitaitiMakoto** [comment]
I rebased and removed `auth_` prefix.

**Comment by kou** [comment]
Thanks. I've merged.

**Comment by KitaitiMakoto** [comment]
Thank you!

**Pull request merged**